executor/ddl.go

```diff
@@ -220,10 +242,40 @@ func (e *DDLExec) Next(ctx context.Context, req *chunk.Chunk) (err error) {
 
 func (e *DDLExec) executeTruncateTable(s *ast.TruncateTableStmt) error {
 	ident := ast.Ident{Schema: s.Table.Schema, Name: s.Table.Name}
+	if _, exist := e.getLocalTemporaryTable(s.Table.Schema, s.Table.Name); exist {
+		return e.executeTruncateLocalTemporaryTable(s)
+	}
 	err := domain.GetDomain(e.ctx).DDL().TruncateTable(e.ctx, ident)
 	return err
 }
 
+func (e *DDLExec) executeTruncateLocalTemporaryTable(s *ast.TruncateTableStmt) error {
+	tbl, exists := e.getLocalTemporaryTable(s.Table.Schema, s.Table.Name)
```

Comment:
We should allow truncate table when database was dropped. So we will not check the exists of database here.